test/SlowTests/Server/Documents/AI/GenAi/Issues/RavenDB-24645.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using FastTests;
+using Microsoft.Azure.Documents.SystemFunctions;
+using Newtonsoft.Json;
+using Raven.Client.Documents.Operations.AI;
+using Raven.Client.Documents.Operations.ConnectionStrings;
+using Raven.Server.Documents;
+using Sparrow.Json;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using Task = System.Threading.Tasks.Task;
+
+namespace SlowTests.Server.Documents.AI.GenAi.Issues
+{
+    public class RavenDB_24645(ITestOutputHelper output) : RavenTestBase(output)
+    {
+        public record Item(ImageDescription[] ImageDescriptions = null);
+
+        public record ImageDescription(string Description, bool SafeForWork, string[] Tags);
+
+        private const string HeartPngBase64 =
+            "iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAADsMAAA7DAcdvqGQAAAGHaVRYdFhNTDpjb20uYWRvYmUueG1wAAAAAAA8P3hwYWNrZXQgYmVnaW49J++7vycgaWQ9J1c1TTBNcENlaGlIenJlU3pOVGN6a2M5ZCc/Pg0KPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyI+PHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj48cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0idXVpZDpmYWY1YmRkNS1iYTNkLTExZGEtYWQzMS1kMzNkNzUxODJmMWIiIHhtbG5zOnRpZmY9Imh0dHA6Ly9ucy5hZG9iZS5jb20vdGlmZi8xLjAvIj48dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPjwvcmRmOkRlc2NyaXB0aW9uPjwvcmRmOlJERj48L3g6eG1wbWV0YT4NCjw/eHBhY2tldCBlbmQ9J3cnPz4slJgLAAAA1ElEQVRYR+2WORLDIAxF5Rwhvcvc/0Ap0+cKpMID4gtJmK3IqzwYfb2Rl+EIIQRayIMvzObQJvA9X9f18/PO7kl4akSBNATBg737I1BAC4vEUOt+AiKFgCesBS6QvYSjmxPosfwruAS42UjSXvtMYBV/gX0E+A9iJGmvfSawikxgxmPgPfaaAAHDnqDsQmA2UACZ3kXKhAJUKWihliUKkFJoRcuoCpAhoIalVhUgYxDHWmMSIEcgOfcWp2IL0vHN0zhinkAKaoTWLDRNoCdNE+jJcoEf1VNdHhBR9pYAAAAASUVORK5CYII=";
+        
+        private const string NonEmptyAnswerHint =
+            " ;Always provide a valid structured response matching the schema (if you have no answer or an empty answer - please return default values instead)";
+
+        private const string JpegScript = @"
+ai.genContext({})
+    .withJpeg(loadAttachment('image.png'));
+";
+
+        private const string PngScript = @"
+ai.genContext({})
+    .withPng(loadAttachment('image.Jpg'));
+";
+
+        private const string ImagePng = "image.png";
+        private const string ImageJpg = "image.Jpg";
+
+        [RavenTheory(RavenTestCategory.Ai)]
+        [RavenGenAiData(IntegrationType = RavenAiIntegration.OpenAi, DatabaseMode = RavenDatabaseMode.Single, CheckCanConnect = false, NightlyBuildRequired = false, Data = new object[] { true })]
+        [RavenGenAiData(IntegrationType = RavenAiIntegration.OpenAi, DatabaseMode = RavenDatabaseMode.Single, CheckCanConnect = false, NightlyBuildRequired = false, Data = new object[]{ false })]
+        public async Task CanUseImageWithTheWrongFormat(Options options, GenAiConfiguration config, bool withJpeg)
```

Comment:
let's make here the params have "string actualFormat, string requestedFromat".
So we can easily see the matrix of what is transparently working with what

Reply:
and maybe let's try other combinations as well?